### PR TITLE
Fix balance related blockers for migration

### PR DESCRIPTION
**Pull request by truemiller**
## Proposed changes
- Fix balance related blockers for migration

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: frontend/components/MainPage/sections/OlasBalanceSection.tsx [hunk 1]
```diff
@@ -36,31 +36,40 @@ export const MainOlasBalance = () => {
...
-    // olas across master wallets, safes and eoa
+    // olas across master wallet -- safes and eoa -- on relevant chains for agent
     const masterWalletOlasBalance = masterWalletBalances?.reduce(
-      (acc, { symbol, balance }) => {
-        if (symbol === TokenSymbol.OLAS) {
+      (acc, { symbol, balance, evmChainId }) => {
+        if (
+          symbol === TokenSymbol.OLAS &&
+          selectedAgentConfig.requiresAgentSafesOn.includes(evmChainId)
+        ) {
```
Comment (truemiller) [inline frontend/components/MainPage/sections/OlasBalanceSection.tsx:45]: fine to aggregate olas across potential chains for display balance

File: frontend/components/MainPage/sections/OlasBalanceSection.tsx [hunk 2]
```diff
@@ -36,31 +36,40 @@ export const MainOlasBalance = () => {
...
-    // olas across all service wallets
+    // olas across all wallets owned by selected service
     const serviceWalletOlasBalance = serviceWalletBalances?.reduce(
-      (acc, { symbol, balance }) => {
-        if (symbol === TokenSymbol.OLAS) {
+      (acc, { symbol, balance, evmChainId }) => {
+        if (
+          symbol === TokenSymbol.OLAS &&
+          selectedAgentConfig.requiresAgentSafesOn.includes(evmChainId)
+        ) {
```
Comment (truemiller) [inline frontend/components/MainPage/sections/OlasBalanceSection.tsx:59]: ensuring we don't calculate funds not on chains irrelevant to the agent

File: frontend/components/ManageStakingPage/StakingContractSection/CantMigrateAlert.tsx [hunk 3]
```diff
@@ -61,24 +59,26 @@ const AlertInsufficientMigrationFunds = ({
...
-  const requiredOlasDeposit = isServiceStaked
-    ? requiredStakedOlas -
-      (totalStakedOlasBalance + safeBalance[TokenSymbol.OLAS]) // when staked
-    : requiredStakedOlas - safeBalance[TokenSymbol.OLAS]; // when not staked
+  const requiredOlasDeposit =
+    requiredStakedOlas -
+    sum([
+      masterSafeBalanceOnHomeChain[TokenSymbol.OLAS],
+      totalStakedOlasBalanceOnHomeChain,
+    ]);
```
Comment (truemiller) [inline frontend/components/ManageStakingPage/StakingContractSection/CantMigrateAlert.tsx:71]: removed this condition, may be breaking due to suspension or eviction
`sum` allows for null/undefined so don't need to guard here
middleware will migrate funds from the current contract if migration required

File: frontend/components/YourWalletPage/index.tsx [hunk 4]
```diff
@@ -69,11 +69,18 @@ const Address = () => {
...
   const masterSafeOlasBalance = masterWalletBalances
-    ?.filter((walletBalance) => walletBalance.symbol === TokenSymbol.OLAS)
+    ?.filter(
+      (walletBalance) =>
+        walletBalance.symbol === TokenSymbol.OLAS &&
+        selectedAgentConfig.requiresMasterSafesOn.includes(
+          walletBalance.evmChainId,
+        ),
+    )
     .reduce((acc, balance) => acc + balance.balance, 0);
```
Comment (truemiller) [inline frontend/components/YourWalletPage/index.tsx:84]: found this confusing as funds from base where shown here while on predict

**Review by Tanya-atatakai — APPROVED** [review]

**Pull request merged**